### DISCUSSION TITLE
Audio recording for the entire experiment?

### DISCUSSION THREAD

**Discussion by zhangchi199133** [jspsych/jsPsych#1161] [Q&A] [answered]
Hi there!

I was wondering whether there is a way to run the audio recording in the background so that everything that is produced throughout the experiment could be recorded? Thank you so much!

Best,

Chi

**Comment by becky-gilbert** [comment, marked as answer]
Hi Chi, you could look at the audio-response plugins here: https://github.com/becky-gilbert/jsPsych/tree/audio-response/plugins to see how to record audio in JavaScript. However these plugins are just for recording audio during a single trial. To get a recording that lasts throughout the experiment, you can look at the plugin code to see how to start the recording at the start of the experiment, and end the recording and save the data at the end of the experiment. You could use the [call-function plugin](https://www.jspsych.org/plugins/jspsych-call-function/) to run arbitrary JavaScript code at any point during your experiment.
